Objects/rangeobject.c

```diff
@@ -979,75 +998,87 @@ typedef struct {
 static PyObject *
 longrangeiter_len(longrangeiterobject *r, PyObject *no_args)
 {
-    Py_INCREF(r->len);
-    return r->len;
+    PyObject *len;
+    Py_BEGIN_CRITICAL_SECTION(r);
```

Comment:
Out of curiosity, using atomic operation is not enough for this operation?

Reply:
It's not enough to make this an atomic read without also making the writes in the other critical sections atomic (in addition to using TryIncrefCompare here); it's still a data race to write non-atomically even if all reads are atomic. I didn't really want to do all that work just to avoid a critical section in this function, when this isn't a particularly common or performance-sensitive function, and an uncontested critical section is probably just as fast anyway.